apps/emqx_authz/include/emqx_authz.hrl

```diff
@@ -17,7 +17,15 @@
 -type(rule() :: {permission(), who(), action(), list(emqx_types:topic())}).
 -type(rules() :: [rule()]).
 
--type(sources() :: [map()]).
+-type(source() :: map()).
```

Comment:
putting type spec to headers are actually against elvis' wishes.
preferred is to export types.

Reply:
🆗 What we better do with existing types?

Reply:
would be nice to refactor if you are there already.

Reply:
I hid most of the types to the modules to which they belong, but some types are used to describe `emqx_acl` record, and this record is purposely shared: it is read from mnesia, stored to ets, etc.

It's quite a big refactoring to get rid of this too, so we better do it in a separate PR (if we want to).